src/runtime/plugin.client.ts

```diff
@@ -71,6 +71,20 @@ export default defineNuxtPlugin((nuxtApp) => {
     })
   }
 
+  function setPreferenceToStorage (storageType: typeof storage, preference: string) {
+    switch (storageType) {
+      case 'cookie':
+        window.document.cookie = storageKey + '=' + preference
```

Comment:
Need ability to pass extra cookie options such as `expires`, `max-age`, `path`, `samesite`, `secure`, etc.